src/core/cluster.py

```diff
@@ -334,3 +375,10 @@ def ready(self) -> Status:
             return Status.ALL_UNIFIED
 
         return self.stable
+
+    def has_peer_relation(self) -> bool:
+        """The cluster has a peer relation."""
+        try:
+            return bool(self.peer_relation)
+        except AttributeError:
+            return False
```

Comment:
This method can be removed and use directly the new boolean override from `RelationState`

Reply:
Good calls, I missed those. Fixed on 34dd6d